MySortedSet/homework4/set/MySortedSet.java

```diff
@@ -0,0 +1,84 @@
+package homework4.set;
+
+/**
+ * Custom implementation of a sorted set.
+ * Contains methods
+ * {@link #add(Comparable) add()},
+ * {@link #remove(Comparable) remove()},
+ * {@link #getSize() getSize()},
+ * {@link #contains(Comparable) contains()}, and
+ * {@link #print() print()} public methods},
+ * <p>
+ * It's a generic class that stores objects in a  Binary Search Tree data structure.
+ * It doesn't allow duplicates or nulls.
+ * The objects are sorted in acsending order.
+ *
+ * @author Nonna Hakobyan
+ */
+
+public class MySortedSet<T extends Comparable<T>> {
+    private MyBinaryTree<T> myTree;
+    private int size;
+
+    public MySortedSet() {
+        this.myTree = new MyBinaryTree<>();
+    }
+
+    /**
+     * Adds new objects to the MySortedSet
+     *
+     * @param object The object to be added to the set. Object must implement Comparable interface
+     * @return  A boolean specifier, which is false if something went wrong and the object was not added
+     *          and is true if object is added to the set successfully
+     */
+    public boolean add(T object) {
```

Comment:
Consider creating a validation method that checks whether the element equals to null and throws a NullPointerException. Use it in add, remove, and contains methods.